webapp/channels/src/components/admin_console/admin_settings/admin_settings.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable react/require-optimization */
+
+import React, {useCallback} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import FormError from 'components/form_error';
+import SaveButton from 'components/save_button';
+import AdminHeader from 'components/widgets/admin_console/admin_header';
+import WithTooltip from 'components/with_tooltip';
+
+type Props = {
+    isDisabled?: boolean;
+    renderTitle: () => JSX.Element;
+    renderSettings: () => React.ReactNode;
+    doSubmit: () => void;
+    saving: boolean;
+    saveNeeded: boolean;
+    serverError?: React.ReactNode;
+}
+
+const AdminSettings = ({
+    doSubmit,
+    renderSettings,
+    renderTitle,
+    isDisabled,
+    saving,
+    saveNeeded,
+    serverError,
+}: Props) => {
+    const handleSubmit = useCallback((e: React.FormEvent<HTMLFormElement> | React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        e.preventDefault();
+
+        doSubmit();
+    }, [doSubmit]);
+
+    return (
+        <form
+            className='form-horizontal'
+            role='form'
+            onSubmit={handleSubmit}
+        >
+            <div className='wrapper--fixed'>
+                <AdminHeader>
+                    {renderTitle()}
+                </AdminHeader>
+                {renderSettings()}
```

Comment:
This comes untouched from https://github.com/mattermost/mattermost/blob/58607eace8ec9b2692775055a132ac3b8f151f3d/webapp/channels/src/components/admin_console/old_admin_settings.tsx#L217

I tried not to make too many extra changes on this, to keep as much as possible the current behavior.

With children, I guess you mean that `AdminSettings` have a children prop, and then is called like...
```
<AdminSettings renderTitle={renderTitle}>
    {renderSettings()}
</AdminSettings>
```
This could work, but I am not a big fan of the children prop. It breaks memos, allows arbitrary JSX and I don't find the code clearer. But that is mostly personal opinion.

That being said, I feel we can revisit this on a future iteration, and leave this just as it was.

Reply:
Huh, turns out I'm the reason that it works that way 😛

https://github.com/mattermost/mattermost/pull/3003/files#diff-a84f50ee948aad519dcebf6824ecdc16706367841d1500bcc2e84e9a47ef462dR100

> This could work, but I am not a big fan of the children prop. It breaks memos, allows arbitrary JSX and I don't find the code clearer. But that is mostly personal opinion.

I know we discussed this yesterday, but if it does break memo, then that's likely because we're overusing memo. [This section of the React docs](https://react.dev/reference/react/memo#should-you-add-memo-everywhere) specifically mentions passing children as JSX making memo unnecessary.

Regardless, yeah, let's leave this as-is